File: superfields/src/main/resources/META-INF/resources/frontend/super-tabs-header.js
import {Tabs} from '@vaadin/tabs';
import {tabsStyles} from '@vaadin/tabs/src/styles/vaadin-tabs-base-styles.js';
import {CSS_SUPER_TABS_MULTILINE, CSS_SUPER_TABS_STYLED_OVERFLOW_BUTTONS} from "./styles/css-super-tabs-multiline";

class SuperTabsHeader extends Tabs {

    static get is() {
        return 'super-tabs-header'
    }

    static get styles() {
        return [tabsStyles, CSS_SUPER_TABS_MULTILINE, CSS_SUPER_TABS_STYLED_OVERFLOW_BUTTONS]
    }

}

customElements.define(SuperTabsHeader.is, SuperTabsHeader);

File: superfields/src/main/java/org/vaadin/miki/superfields/tabs/SuperTabsHeader.java
package org.vaadin.miki.superfields.tabs;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.tabs.Tabs;

/**
 * An extension of {@link Tabs} with some additional styles added in the js file.
 * This class serves no other purpose and probably should not be used outside of {@link SuperTabs}.
 *
 * @author miki
 * @since 2026-01-18
 */
@JsModule("./super-tabs-header.js")
@Tag("super-tabs-header")
public class SuperTabsHeader extends Tabs {
}
